app/main.py

```diff
@@ -1,8 +1,57 @@
+from typing import List
+
+
 class Car:
-    # write your code here
-    pass
+    def __init__(
+        self,
+        comfort_class: int,
+        clean_mark: int,
+        brand: str
+    ) -> None:
+        self.comfort_class: int = comfort_class
+        self.clean_mark: int = clean_mark
+        self.brand: str = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(
+        self,
+        distance_from_city_center: float,
+        clean_power: int,
+        average_rating: float,
+        count_of_ratings: int,
+    ) -> None:
+        self.distance_from_city_center: float = distance_from_city_center
+        self.clean_power: int = clean_power
+        self.average_rating: float = average_rating
+        self.count_of_ratings: int = count_of_ratings
+
+    def calculate_washing_price(self, car: Car) -> float:
+        price = (
+            car.comfort_class
+            * (self.clean_power - car.clean_mark)
+            * self.average_rating
+            / self.distance_from_city_center
+        )
+        return round(price, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        if car.clean_mark < self.clean_power:
+            car.clean_mark = self.clean_power
+
+    def serve_cars(self, cars: List[Car]) -> float:
+        total_income = 0.0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                price = self.calculate_washing_price(car)
+                total_income += price
+                self.wash_single_car(car)
+        return round(total_income, 1)
+
+    def rate_service(self, rating: int) -> None:
```

Comment:
Ensure that the `rating` parameter is within a valid range (e.g., 1 to 5) before updating the average rating. This will prevent invalid ratings from affecting the average.